backend/src/email/middlewares/email-transactional.middleware.ts

```diff
@@ -212,6 +221,31 @@ export const InitEmailTransactionalMiddleware = (
     res.status(200).json(convertMessageModelToResponse(message))
   }
 
+  async function listMessages(req: Request, res: Response): Promise<void> {
+    // validation from Joi doesn't carry over into type safety here
+    // following code transforms query params into type-safe arguments for EmailTransactionalService
+    const { limit, offset, status, created_at, sort_by } = req.query
+    const userId: string = req.session?.user?.id.toString() // id is number in session; convert to string for tests to pass (weird)
+    const filter = created_at ? { createdAt: created_at } : undefined
+    const sortBy = sort_by?.toString().replace(/[+-]/, '')
+    const orderBy = sort_by?.toString().includes('+')
+      ? Ordering.ASC
+      : Ordering.DESC // default to descending order even without '-' prefix
+    const { hasMore, messages } = await EmailTransactionalService.listMessages({
+      userId,
+      limit: +(limit as string),
+      offset: +(offset as string),
+      sortBy: sortBy as TransactionalEmailSortField,
+      orderBy,
+      status: status as TransactionalEmailMessageStatus,
+      filterByTimestamp: filter as unknown as TimestampFilter,
```

Comment:
is this as unknown necessary? tried it on my IDE it works without as unkown.

Reply:
ohh weird you're right it works now (I'm almost sure I added it because I was prompted by my IDE).

good catch, have removed it.